src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -0,0 +1,180 @@
+import _isEmpty from 'lodash/isEmpty';
+import _isObject from 'lodash/isObject';
+import _map from 'lodash/map';
+import LegModel from './LegModel';
+import { routeStringFormatHelper } from '../../navigationLibrary/Route/routeStringFormatHelper';
+
+/**
+ *
+ * This class should always be instantiated from an `AircraftInstanceModel` and
+ * always instantiated with some form of a `spawnPatternModel`.
+ *
+ * @class Fms
+ */
+export default class Fms {
+    /**
+     * @constructor
+     * @param aircraftInitProps {object}
+     * @param initialRunwayAssignment {string}
+     * @param navigationLibrary {NavigationLibrary}
+     */
+    constructor(aircraftInitProps, initialRunwayAssignment, navigationLibrary) {
+        if (!_isObject(aircraftInitProps) || _isEmpty(aircraftInitProps)) {
+            throw new TypeError('Invalid aircraftInitProps passed to Fms');
+        }
+
+        /**
+         *
+         *
+         * @property _navigationLibrary
+         * @type {NavigationLibrary}
+         * @private
+         */
+        this._navigationLibrary = navigationLibrary;
+
+        /**
+         *
+         *
+         * @property _runway
+         * @type {string}
+         * @private
+         */
+        this._runway = initialRunwayAssignment;
+
+        /**
+         *
+         *
+         * @property legCollection
+         * @type {array}
+         * @default []
+         */
+        this.legCollection = [];
+
+        /**
+         *
+         *
+         * @property category
+         * @type {string}
+         * @default ''
+         */
+        this.category = '';
+
+        this.init(aircraftInitProps);
+    }
+
+    /**
+     *
+     *
+     * @property currentWaypoint
+     * @return {FixModel|StandardRouteWaypointModel|undefined}
```

Comment:
good catch here, an earlier implementation did not yet have the new `WaypointModel` so I was working with the classes listed there.

The short answer is no.  the `StandardRouteWaypointModel` give us functionality that the WaypointModel does not.  It has a very specific concern that I don't think we should generalize.  I _do_ think it could be simplified slightly, but I don't see a reason to get rid of it anytime soon.